src/application_config/configuring_base.py

```diff
@@ -21,49 +22,45 @@
 
 
 @dataclass(frozen=True)
-class ConfigSectionBase(ABC):
-    """Base class for all ConfigSection classes"""
+class ConfigSectionBase:
+    """Base class for all ConfigSection classes (so that we can bound a TypeVar)"""
 
 
-# The next line has a type: ignore because MyPy has a problem with @abstractmethod
-# If you comment the line with @abstractmethod, you can do type checking
 @dataclass(frozen=True)
-class ConfigBase(ABC):
+class ConfigBase:
     """Base class for main Config class"""
 
-    @staticmethod
-    @abstractmethod
-    def get_app_basename() -> str:
-        """Return the application base name.
-
-        This name, with a preceding dot, will also be the folder name in the home directory
-        that will store the config.
-        """
+    @classmethod
+    def default_config_foldername(cls: type[TConfig]) -> str:
```

Comment:
Do we want to allow this method to return `None`? That way a user could signal that you **have** to supply a configfolder_path in the `get`.

We could add some custom exception if we cannot get a valid default filepath.

Reply:
Then we should rather allow "default_config_filepath" to return None. Will change that.